src/DynamoCore/Core/UndoRedoRecorder.cs

```diff
@@ -363,17 +382,37 @@ private void UndoActionGroup(XmlElement actionGroup)
             for (int index = actions.Count - 1; index >= 0; index--)
             {
                 var element = actions[index] as XmlElement;
+
                 XmlAttribute actionAttribute = element.Attributes[UserActionAttrib];
                 var modelActionType = (UserAction)Enum.Parse(typeof(UserAction), actionAttribute.Value);
+
                 switch (modelActionType)
                 {
                     // Before undo takes place (to delete the model), the most 
                     // up-to-date model is retrieved and serialized into the 
                     // redo action group so that it can properly be redone later.
                     case UserAction.Creation:
-                        ModelBase toBeDeleted = undoClient.GetModelForElement(element);
-                        RecordActionInternal(newGroup, toBeDeleted, modelActionType);
-                        undoClient.DeleteModel(element);
+                        try
+                        {
+                            ModelBase toBeDeleted = undoClient.GetModelForElement(element);
+                            RecordActionInternal(newGroup, toBeDeleted, modelActionType);
+                            undoClient.DeleteModel(element);
+                        }
+                        catch (ArgumentException e)
+                        {
+                            bool isOffTrackObject = false;
+                            XmlAttribute guidAttribute = element.Attributes["guid"];
+                            if (null != guidAttribute)
+                            {
+                                var guid = Guid.Parse(guidAttribute.Value);
+                                isOffTrackObject = offTrackModels.Contains(guid);
+                            }
+
+                            if (!isOffTrackObject)
+                            {
+                                throw e;
+                            }
+                        }
```

Comment:
A small thing, I noticed you stopped using implicitly typed `var` for a while now :)